python/cudf/dataframe/dataframe.py

```diff
@@ -235,8 +235,13 @@ def __getitem__(self, arg):
                 mask = np.array(mask)
             df = DataFrame()
             if(mask.dtype == 'bool'):
+                # New df-wide index
+                selvals, selinds = columnops.column_select_by_boolmask(
+                        columnops.as_column(self.index), Series(mask))
+                index = self.index.take(selinds.to_gpu_array())
                 for col in self._cols:
-                    df[col] = self._cols[col][arg]
+                    df[col] = Series(self._cols[col][arg], index=index)
```

Comment:
I think we should fix this more generally, where columns assigned to a dataframe via __setitem__ shouldn't be able to have an index other than the dataframe's index. Does that behavior sound correct / expected?

Reply:
I guess I'm asking to tackle the more general `__setitem__` case instead of fixing the symptom here.

Reply:
You're thinking that any time a Series is added to a new dataframe, it should always receive the identical index as its parent? Given that Indexes are immutable, I completely agree. That's a more challenging issue to pile onto this one though, create a new issue for it?